### ISSUE TITLE
Git Variables Not Expanded in Error Emails

### ISSUE THREAD

**Issue by sjv0** [woolfg/dokuwiki-plugin-gitbacked#78]
Hi woolfg, thanks for creating such a useful plugin!

I'm still trying to get it to work, but I'm getting an error email that doesn't show the actual error that was encountered:

```
An FAILURE with the Git repo path occurred on http://my.dokuwiki.site.com/

Repo path: <code>@GITREPOPATH@</code>
Error: <code>@GITERRORMESSAGE@</code>
User:
My Name
eMail:
my.email@example.com
Browser:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.5 Safari/605.1.15
Wiki:
My Wiki Name http://my.dokuwiki.site.com/
```

I have the following settings defined:

```
$ egrep 'gitbacked|datadir|mediadir' /usr/local/www/nginx/dokuwiki/conf/local.php
$conf['plugin']['gitbacked']['pushAfterCommit'] = 1;
$conf['plugin']['gitbacked']['repoPath'] = '/usr/local/dev/main/';
$conf['plugin']['gitbacked']['repoWorkDir'] = '/usr/local/dev/main/docs/dokuwiki/data/';
$conf['plugin']['gitbacked']['gitPath'] = '/usr/local/bin/git';
$conf['plugin']['gitbacked']['addParams'] = '-c user.email="<%mail%>" -c user.name="%user%"';
$conf['plugin']['gitbacked']['emailAddressOnError'] = 'my.email@example.com';
$conf['plugin']['gitbacked']['notifyByMailOnSuccess'] = 1;
$conf['datadir'] = '/usr/local/dev/main/docs/dokuwiki/data/pages';
$conf['mediadir'] = '/usr/local/dev/main/docs/dokuwiki/data/media';
```

I've confirmed that all the paths are correct.  `/usr/local/bin/git` is where my git executable is, `/usr/local/dev/main/` is where my existing repo is with a .git directory inside it, and `/usr/local/dev/main/docs/dokuwiki/data/` is where I moved my pages and media directories.

I've set all the necessary permissions to allow the www user to make changes to files in the git repo, add them and commit them.  I've also set up the remote repository to allow the www user to push changes to it.

If I change a document in the wiki, I get the above error email.  The change appears in the git repo as a modified file as expected, so my datadir setting is working.  The modified file can be manually committed as the www user:

```
sudo -u www git status
sudo -u www git add
sudo -u www git commit
sudo -u www git push
```

So I feel like I've set everything up as per the documentation but I don't know how to find out what's triggering the error.  Do you know why it might be that the `@GITREPOPATH@` and `@GITERRORMESSAGE@` variables in the email aren't getting expanded?

Thank you very much for your time,

Sam

**Comment by sjv0** [comment]
I've done some more digging and found:

1. The GIT_REPO_PATH and GIT_ERROR_MESSAGE variables aren't working because the underscores are converted to <em> and </em> tags in the HTML before the template variables are replaced.  This is why all the other variables that don't have underscores worked.
2. The gitbacked code is appending the supplied repo paths to the dokuwiki dir, so they have to be specified as relative paths to that directory.  It would be good if the code could be changed to treated them as absolute paths if they start with a slash.  Either that or make it clear in the configuration that they have to be relative paths.

I can now get the git commands to work by using relative paths.  Thanks again for the plugin.

**Comment by sjv0** [comment]
OK I'll update the documentation but the fix for issue #1 above is going to require a code change.  Either the variables in the templates will all need their underscores removed (and corresponding changes made to the code), or a change will need to be made to prevent the templates from being processed as Wiki text.

**Comment by sjv0** [comment]
Documentation updated as requested, but lang/en/settings.php should really be updated as well.

Even better, check the first character of the path in the code and treat as an absolute path if it starts with a '/'.  (don't prepend the savedir)

**Comment by woolfg** [comment]
thanks @sjv0 for the update. we also appreciate pull requests to change this part of the code.

**Comment by sjv0** [comment]
I'd love to help but unfortunately I've never written a line of PHP in my life.  Probably better to leave it to the experts!

**Comment by woolfg** [comment]
I see ;) could you summarize the problem and what should be done in a new issue, so anyone can pick it up and solve it....
